src/CMakeLists.txt

```diff
@@ -14,8 +14,8 @@
 
 # Configure config.hpp
 set(PROBLEM_GENERATOR "<not-implemented>") # TODO: Figure out what to put here
-
-if (ENABLE_MPI)
+if ("${ENABLE_MPI}")
```

Comment:
This should be functionally equivalent to what was already there.

Reply:
I think it is better practice to wrap variables in ${}, if ENABLE_MPI="Off" then 
```
if(ENABLE_MPI)
```
Will evaluate to true. This has caused me headaches in the past.

Reply:
?

```
~/Code/cmake-playground via 🛆 v3.18.2 
at 11:03:38 ❯ cat CMakeLists.txt 
cmake_minimum_required(VERSION 3.16)

project(cmake-playground LANGUAGES NONE)

set(ENABLE_MPI "Off")
if (ENABLE_MPI)
    message(STATUS "Yes MPI")
else()
    message(STATUS "No MPI")
endif()%                                                                                                                                               

~/Code/cmake-playground via 🛆 v3.18.2 
at 11:03:45 ❯ cmake -S. -Bbuild
-- No MPI
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/agaspar/Code/cmake-playground/build
```

Reply:
Regardless, I don't see why changing it from `if(<variable>)` to `if(${<variable>})` would change whether cmake would compare the value as case insensitive.

Reply:
Doesn't seem to be a version issue - I tested on CMake 3.12 (the minimum supported CMake version) and "Off" is falsey there, too.

Reply:
Ah, I stand corrected, I don't know what problem I ran into in the past, I will correct this.

Reply:
It actually looks like not including ${} is better practice.

Reply:
Didn't look like this was resolved yet? Or maybe you just hadn't pushed the changes yet.

Reply:
Ya, I just haven't pushed.